lib.js

```diff
@@ -59,51 +57,31 @@ export async function fetchInUseAMIIDs(ec2, autoscaling) {
       ]
     }]
   });
-  for await (const page of paginator) {
+  for await (const page of instancePaginator) {
     for (const reservation of page.Reservations) {
       reservation.Instances.forEach(instance => inUseAMIIDs.add(instance.ImageId));
     }
   }
 
-  const asgs = [];
-  for await (const asg of (async function*() {
-    let nextToken = '';
-    while (nextToken !== undefined) {
-      const {AutoScalingGroups, NextToken} = await autoscaling.send(new DescribeAutoScalingGroupsCommand({
-        NextToken: (nextToken === '') ? undefined : nextToken
-      }));
-      yield* AutoScalingGroups;
-      nextToken = NextToken;
-    }
-  })()) {
-    asgs.push(asg);
+  const lcPaginator = paginateDescribeLaunchConfigurations({client: autoscaling}, {});
+  for await (const page of lcPaginator) {
+    page.LaunchConfigurations.forEach(lc => inUseAMIIDs.add(lc.ImageId));
   }
 
-  // in use by ASG -> Launch Configuration
-  const inUseLCNames = asgs.filter(asg => 'LaunchConfigurationName' in asg).map(asg => asg.LaunchConfigurationName);
-  if (inUseLCNames.length > 0) {
-    for (let i = 0; i < Math.ceil(inUseLCNames.length/MAX_ITEMS_PER_LAUNCH_CONFIGURATION_PAGE); i++) {
-      const {LaunchConfigurations: lcs} = await autoscaling.send(new DescribeLaunchConfigurationsCommand({
-        LaunchConfigurationNames: inUseLCNames.slice(i*MAX_ITEMS_PER_LAUNCH_CONFIGURATION_PAGE, (i+1)*MAX_ITEMS_PER_LAUNCH_CONFIGURATION_PAGE)
-      }));
-      lcs.forEach(lc => inUseAMIIDs.add(lc.ImageId));
-    }
+  const ltPaginator = paginateDescribeLaunchTemplates({client: ec2}, {});
+  const ltLimit = pLimit(5);
+  for await (const page of ltPaginator) {
+    await Promise.all(
+      page.LaunchTemplates.map(({LaunchTemplateId: id, DefaultVersionNumber: version}) => ltLimit(async () => { 
```

Comment:
@andreaswittig each Launch Template can have multiple versions. Each version links to an AMI. Would you cheget the AMI IDs from all versions (maximum is 10,000 versions), just the default version (as implemented), or the latest version?

Reply:
@michaelwittig Using the default version seems legit.